validator.py

```diff
@@ -1543,6 +1548,104 @@ def validate_evac_ids(self):
+            missing = 0
+            for event in events:
+                source = event.get('source', None)
+                obj = event.get('object', None)
+                if source is None:
+                    missing += 1
+                elif source not in entity_type_enum:
+                    in_any_group = False
+                    in_all_groups = True
+                    for group in chars['possible']:
+                        if source in group:
+                            in_any_group = True
+                        else:
+                            in_all_groups = False
+                    if not in_any_group:
+                        self.logger.log(LogLevel.ERROR, f"The 'source' parameter for an event in scene '{scene['id']}' is '{source}', but must be one of {entity_type_enum + chars['possible']}.")
+                        self.invalid_values += 1 
+                    elif not in_all_groups:
+                        self.logger.log(LogLevel.WARN, f"The 'source' parameter for an event in scene '{scene['id']}' is '{source}', but that character might not be available in some branches.")
+                        self.warning_count += 1     
+                    elif source in chars['removed']:
+                        self.logger.log(LogLevel.WARN, f"The 'source' parameter for an event in scene '{scene['id']}' is '{source}', but that character might be removed in some branches.")
+                        self.warning_count += 1 
+
+                if obj is not None and obj not in entity_type_enum:
+                    in_any_group = False
+                    in_all_groups = True
+                    for group in chars['possible']:
+                        if source in group:
+                            in_any_group = True
+                        else:
+                            in_all_groups = False
+                    if not in_any_group:
+                        self.logger.log(LogLevel.ERROR, f"The 'object' parameter for an event in scene '{scene['id']}' is '{obj}', but must be one of {entity_type_enum + chars['possible']}.")
+                        self.invalid_values += 1 
+                    elif not in_all_groups:
+                        self.logger.log(LogLevel.WARN, f"The 'object' parameter for an event in scene '{scene['id']}' is '{obj}', but that character might not be available in some branches.")
+                        self.warning_count += 1     
+                    elif source in chars['removed']:
+                        self.logger.log(LogLevel.WARN, f"The 'object' parameter for an event in scene '{scene['id']}' is '{obj}', but that character might be removed in some branches.")
+                        self.warning_count += 1 
+            if missing > 0:
+                self.logger.log(LogLevel.WARN, f"The 'object' parameter is recommended for all events, but is missing for {missing} event{'s' if missing > 1 else ''} in scene '{scene['id']}'.")
```

Comment:
Essentially a typo:  object parameter should be source.

Reply:
silly me